README.md

```diff
@@ -1,4 +1,262 @@
+are given an `args` variable. `args` is a Python dictionary. Both keys and values for this Python dictionary are strings. You can find the available keys in the `args` dictionary along with their description in the table below:
+
+| key                      | description                                      |
+|--------------------------|--------------------------------------------------|
+| model_config             | A JSON string containing the model configuration |
+| model_instance_kind      | A string containing model instance kind          |
+| model_instance_device_id | A string containing model instance device ID     |
+| model_repository         | Model repository path                            |
+| model_version            | Model version                                    |
+| model_name               | Model name                                       |
+
+### `execute`
+
+`execute` function is called whenever an inference request is made. Every Python
+model **MUST** implement `execute` function. In the `execute` function you are given
+a list of `InferenceRequest` objects. In this fucntion, your `execute` function
+must return a list of `InferenceResponse` objects that has the same length as 
+`requets`.
+
+In case one of the inputs has an error, you can use the `TritonError` object to set the error message for that specific request. Below is an example of setting errors for an `InferenceResponse` object:
+
+```python
+import triton_python_backend_utils as pb_utils
+
+
+class TritonPythonModel:
+    ...
+
+    def execute(self, requests):
+        responses = []
+
+        for request in requests:
+            if an_error_occured:
+              # If there is an error, the output_tensors are ignored
+              responses.append(pb_utils.InferenceResponse(
+                output_tensors=[], error=pb_utils.TritonError("An Error Occured")))
+
+        return responses
+```
+
+### `finalize`
+
+Implementing `finalize` is optional. This function allows you to do any clean ups necessary before the model is unloaded from Triton server.
+
+You can look at the [add_sub example](examples/add_sub.py) which contains
+a complete example of implementing all these functions for a Python model
+that adds and subtracts the inputs given to it. After implementing all the necessary functions, you should save this file as `model.py`.
+
+## Model Config File
+
+Every Python Triton model must provide a `config.pbtxt` file describing 
+the model configuration. In order to use this backend you must set the `backend`
+field of your model `config.pbtxt` file to `python`. You shouldn't set
+`platform` field of the configuration.
+
+Also, you need to make a copy of [triton_python_backend_utils.py](src/resources/triton_python_backend_utils.py) available to your `model.py`.
+
+Your models directory should look like below:
+```
+models
+├── add_sub
+│   ├── 1
+│   │   └── model.py
+│   └── config.pbtxt
+└── triton_python_backend_utils.py
```

Comment:
This level of the model repository should only have the model subdirs. We should show best practice as having triton_python_backend_utils.py in the same directory as model.py.